content/articles/2019-03-ggplot2-3-2-0.Rmarkdown

```diff
@@ -0,0 +1,116 @@
+
+```{r, warning=FALSE}
+library(ggplot2)
+library(tibble)
+radians <- seq(0, 2 * pi, length.out = 101)[-1]
+circle <- tibble(
+  x = c(cos(radians), cos(radians) * 0.5),
+  y = c(sin(radians), sin(radians) * 0.5),
+  subgroup = rep(c(1, 2), each = 100)
+)
+ggplot(circle) + 
+  geom_polygon(
+    aes(x = x, y = y, subgroup = subgroup), 
+    fill = "firebrick", 
+    colour = "black"
+  )
+```
+
+The other bigger new feature is the ability to modify the guide representation
+of a layer through a new `key_glyph` argument in all geoms. While the defaults
+are often fine, there are situations where a different look serves the 
+visualisation:
+
+```{r}
+ggplot(economics, aes(date, psavert, color = "savings rate")) + 
+  geom_line(key_glyph = "timeseries")
+```
+
+`geom_rug` has seen a range of improvements to give users more control over the appearance of the rug lines. Now, both the length, as well as whether to position them inside or outside of the plot region can be controlled:
+
+```{r}
+ggplot(mtcars, aes(x = wt, y = mpg)) +
+  geom_point() + 
+  geom_rug(sides = "tr", length = unit(1, "mm"), outside = TRUE) + 
+  # Need to turn clipping off if rug is outside plot area
+  coord_cartesian(clip = "off")
+```
+
+Aesthetics will now accept a function returning `NULL` and treat it as setting the aesthetic to `NULL`. This can make it easier to program with ggplot2 e.g. by catching errors from non-existing variables:
+
+```{r}
+df <- data.frame(x = 1:10, y = 1:10)
+wrap <- function(...) tryCatch(..., error = function(e) NULL)
+
+ggplot(df, aes(x, y, colour = wrap(no_such_column))) +
+  geom_point()
+```
+
+Lastly, the `stat_function()` gains the ability to accept purrr-style lambda functions. The use of the formula notation for creating lambda functions has become widespread and it is only natural that ggplot2 accepts it as well:
+
+```{r}
+df <- data.frame(x = 1:10, y = (1:10)^2)
+ggplot(df, aes(x, y)) + 
+  geom_point() +
+  stat_function(fun = ~ .x^2)
+```
+
+## Minor fixes and improvements
+`coord_sf` now behaves as the other coords in relation to how it draws grid lines. This means that the aesthetics of the grid matches that of the other coordinate systems, and that it can be turned off through the theming system. You may see slight visual changes when using `geom_sf` as the default grid lines was slightly thicker in `coord_sf` before this change.
```

Comment:
```suggestion
[`coord_sf`](https://ggplot2.tidyverse.org/reference/ggsf.html) now behaves the same as the other coords in relation to how it draws grid lines. This means that the aesthetics of the grid matches that of the other coordinate systems, and that it can be turned off through the theming system. You may see slight visual changes when using `geom_sf()` as the default grid lines were slightly thicker in `coord_sf()` prior to this change.
```